app/src/main/java/it/niedermann/owncloud/notes/model/SingleNoteWidgetFactory.java

```diff
@@ -48,57 +54,56 @@ public int getViewTypeCount() {
 
     @Override
     public void onDataSetChanged() {
+        SharedPreferences sharedprefs = PreferenceManager.getDefaultSharedPreferences(mContext);
+        long noteID = sharedprefs.getLong(SingleNoteWidget.WIDGET_KEY + mAppWidgetId, -1);
 
+        if (noteID >= 0) {
+            if (db.getNote(noteID) == null) {
+                Log.e(TAG, "Error: note not found");
+                note = null;
+            } else {
+                note = db.getNote(noteID);
+            }
+        }
     }
 
     @Override
     public void onDestroy() {
-
+        db.close();
```

Comment:
Since `NoteSQLiteOpenHelper` is currently a singleton, it's dangerous to close the database from here. Theoretically, it's possible that another objects still wants to use the database. Hence, I suggest to remove this.